src/core/lib/transport/transport.h

```diff
@@ -198,6 +198,8 @@ struct grpc_transport_stream_op_batch_payload {
         grpc_chttp2_grpc_status_to_http2_error. Send a RST_STREAM with this
         error. */
   struct {
+    // Error contract: the recipient of cancel_error must cause cancel_error
```

Comment:
That was my original thought, but this particular one can also go to grpc_transport_stream_op_finish_batch_with_failure which never makes it down to the `ext/transport` . But I guess a `lib/transport` is still part of the transport so it's still legitimate. I can change that.